web-ng/src/app/components/layer-dialog/layer-dialog.component.ts

```diff
@@ -132,12 +121,36 @@ export class LayerDialogComponent implements OnDestroy {
     });
   }
 
-  onProjectLoaded(project: Project) {
-    if (this.layerId === ':new') {
+  initProject(
+    projectId: string,
+    createLayer: boolean,
+    layers?: Map<string, Layer>
+  ) {
+    this.projectId = projectId;
+    if (createLayer) {
       this.layerId = this.dataStoreService.generateId();
+      const fieldId = this.dataStoreService.generateId();
+      this.fields = this.fields.push(
+        new Field(
+          fieldId,
+          FieldType.TEXT,
+          StringMap({
+            en: '',
+          }),
+          /* required= */
+          false,
+          /* index= */
+          0,
+          undefined
+        )
+      );
       this.layer = new Layer(this.layerId, /* index */ -1);
     } else {
-      this.layer = project.layers.get(this.layerId);
+      this.layer = layers?.get(this.layerId);
```

Comment:
Based on previous comments, this logic would go away in favor of throwing an error if the component is called with both `createNew` and `layer`, or with `!createNew` and `!layer`.